src/pkg/services/m365/api/users.go

```diff
@@ -453,50 +458,52 @@ func validateUser(item any) (models.Userable, error) {
 	return m, nil
 }
 
-func toString(ctx context.Context, data any, mailBoxErr *clues.Err) string {
-	dataPointer, ok := data.(*string)
-	if !ok {
-		logger.Ctx(ctx).Info("error getting data from mailboxSettings")
-
-		*mailBoxErr = *ErrMailBoxSettingsNotFound
+func toString(ctx context.Context, key string, data map[string]any) (string, error) {
+	ctx = clues.Add(ctx, "setting_name", key)
 
-		return ""
+	if len(data) == 0 {
```

Comment:
if we just want to check the size of map we can check it at a central place. Taking example of `additionalData` - in current scenario the size of same map is getting checked 6 7 times.